### PR TITLE
Set fast math flags on more FP ops in the SPIRV reader.

### PR DESCRIPTION
**Pull request by jayfoad**

### CODE REVIEW

File: translator/lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -1528,6 +1529,32 @@ Value *SPIRVToLLVM::transConvertInst(SPIRVValue *BV, Function *F,
   }
 }
 
+// Decide whether to set fast math flags on Val, which is usually a newly create
+// instruction. Decorations on BV may prevent us from setting some flags.
+void SPIRVToLLVM::maybeSetFastMathFlags(Value *Val, SPIRVValue *BV) {
```
Comment (linqun) [inline translator/lib/SPIRV/SPIRVReader.cpp]: I prefer remove maybe in function name, looks good for others.

Comment (amdrexu) [reply translator/lib/SPIRV/SPIRVReader.cpp]: I think setFastMathFlags() is enough.

Comment (jayfoad) [reply translator/lib/SPIRV/SPIRVReader.cpp]: Done.

Comment (amdrexu) [reply translator/lib/SPIRV/SPIRVReader.cpp]: Please merge the two commits. Theoretically, we make one commit in one pull request. So you can just update the original commit based on review comments.

Comment (jayfoad) [reply translator/lib/SPIRV/SPIRVReader.cpp]: Done. (I thought you would "squash and merge" the pull request.)

Comment (amdrexu) [reply translator/lib/SPIRV/SPIRVReader.cpp]: Oh, author takes this responsibility.

**Comment by JaxLinAMD** [comment]
retest it please

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit d1d23e7b276d0bac12f9f5a0e17e5862fda1ddb2
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev ffb21e86238819817075e252ad9b49ea5284968d</li>
<li>SPVGEN: dev 051aaa46272aae7808129b98fd9da49f3229eb78</li>
<li>XGL: dev e4b6a2dbeb55b4e964531bd5be0be673dd4321c2</li>
<li>LLVM: amd-vulkan-dev d9e4934189845f30428cad6da16e8745540291db</li>
</ul></details>

<details><summary>Build Failed</summary><ul>


</ul></details>

**Comment by JaxLinAMD** [comment]
@jayfoad  test is failed, could you try to rebase?

**Comment by jayfoad** [comment]
> @jayfoad test is failed, could you try to rebase?

OK, I've updated the tests.

**Comment by JaxLinAMD** [comment]
ok to test

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 185cf3627e7e1db75254e0a101da03df62766f06
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev ffb21e86238819817075e252ad9b49ea5284968d</li>
<li>SPVGEN: dev 051aaa46272aae7808129b98fd9da49f3229eb78</li>
<li>XGL: dev e4b6a2dbeb55b4e964531bd5be0be673dd4321c2</li>
<li>LLVM: amd-vulkan-dev d9e4934189845f30428cad6da16e8745540291db</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 918/918 (100.0%)</li>
<li>Failed: 0/918 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366312/366312 (100.0%)</li>
<li>Failed: 0/366312 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366312/366312 (100.0%)</li>
<li>Failed: 0/366312 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by jayfoad** [comment]
Ping?

**Comment by JaxLinAMD** [comment]
sorry for the delay

**Pull request merged**